deploy/deploy-with-olm.yaml

```diff
@@ -0,0 +1,44 @@
+---
+## Add a custom CatalogSource for the ocs-operator CSV
+apiVersion: operators.coreos.com/v1alpha1
+kind: CatalogSource
+metadata:
+  name: ocs-catalogsource
+  namespace: openshift-marketplace
+spec:
+  sourceType: grpc
+  image: quay.io/ocs-dev/ocs-registry:latest
```

Comment:
at some point in the future, it would be nice if we could generate some of these manifests from templates, and make things like "image" a variable.

Then we could run a make target like `make manifests` which generates deploy/deploy-with-olm.yaml from user env-vars. This will help with creating release manifests as well.